test/e2e/page-objects/pages/test-dapp.ts

```diff
@@ -744,14 +744,17 @@ class TestDapp {
 
   async clickAddTokenToWallet() {
     await this.driver.clickElement(this.addTokensToWalletButton);
+    await this.driver.waitForWindowWithTitleToBePresent(WINDOW_TITLES.Dialog);
```

Comment:
ℹ️ First I just added a wait for X windows on the spec level, to fix that specific test:

```
await testDapp.clickPersonalSign();
await driver.waitUntilXWindowHandles(3);
```
Later, I decided a better approach would be to fix that issue at the test dapp level, so all test benefit from this mitigation.
This uses a new method `waitForWindowWithTitle` which waits until there's a window with the title Dialog.
This *doesn't change the window focus* and each test can do whatever they need, but it ensures that before proceeding with any action, the Dialog is open, as a result of the click action.

This works for any case: even if I had already a dialog open. So this is backwards compatible with any test, and just adds robustness to all the test dapp methods